Introduce global, per-tenant flags and interval to evaluation threshold to control rule evaluation concurrency (#8146)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Expose a new `s3.trace.enabled` configuration option to enable detailed logging of operations against S3-compatible object stores. #8690
* [ENHANCEMENT] memberlist: locally-generated messages (e.g. ring updates) are sent to gossip network before forwarded messages. Introduced `-memberlist.broadcast-timeout-for-local-updates-on-shutdown` option to modify how long to wait until queue with locally-generated messages is empty when shutting down. Previously this was hard-coded to 10s, and wait included all messages (locally-generated and forwarded). Now it defaults to 10s, 0 means no timeout. Increasing this value may help to avoid problem when ring updates on shutdown are not propagated to other nodes, and ring entry is left in a wrong state. #8761
* [ENHANCEMENT] Querier: allow using both raw numbers of seconds and duration literals in queries where previously only one or the other was permitted. For example, `predict_linear` now accepts a duration literal (eg. `predict_linear(..., 4h)`), and range vector selectors now accept a number of seconds (eg. `rate(metric[2])`). #8780

> * [ENHANCEMENT] Ruler: Add `-ruler.max-global-rule-evaluation-concurrency` to allow independent rules of a tenant to be run concurrently. The amount of concurrency per tenant is controlled via `-ruler.max-concurrent-rule-evaluations-per-tenant` as a limit. A `-ruler.max-global-rule-evaluation-concurrency` value of `0` can be used to disable the feature for all tenants. By default, the new behaviour is disabled. A rule is eligible for concurrency as long as it has no rules it depends on nor any rules that depend on it, and its total rule group runtime exceeds 50% of its interval. #8146

Reviewer: ```suggestion
* [ENHANCEMENT] Ruler: Add `-ruler.max-global-rule-evaluation-concurrency` to allow independent rules of a tenant to be run concurrently. You can control the amount of concurrency per tenant via the `-ruler.max-concurrent-rule-evaluations-per-tenant` limit. Use a `-ruler.max-global-rule-evaluation-concurrency` value of `0` to disable the feature for all tenants. By default, this behavior is disabled. A rule is eligible for concurrency as long as it doesn't depend on any other rules, doesn't have any other rules that depend on it, and has a total rule group runtime that exceeds 50% of its interval. #8146
```

Reviewer: We can update this a bit to use active voice.

Author: I haven't forgotten about this. I'll update it once I get confirmation of the flag names and metrics from @pracucci.

Author: Done in https://github.com/grafana/mimir/pull/8146/commits/c4f16e748b7e8d28b45645f82a025ba999ec5a4a

>   * As part of this work the following metrics have been introduced:

Reviewer: ```suggestion
  * This work introduces the following metrics:
```

Reviewer: Minor update for active voice :)
